Dockerfile

```diff
@@ -57,14 +57,52 @@ conda config --show-sources
 conda list --show-channel-urls
 
 # Install RAPIDS
-mamba install -y -n base \
+mamba create -y -n rapids \
     "rapids=${RAPIDS_VER}.*" \
     "python=${PYTHON_VER}.*" \
     "cuda-version=${CUDA_VER%.*}.*" \
     ipython
+
 conda clean -afy
 EOF
 
+# remove the 'conda activate base' in .bashrc that comes from miniforge-cuda,
+# to avoid that happening in the entrypoint script
+RUN sed -i.bak '/conda activate base/d' ~/.bashrc \
+ && rm -f ./*.bak
+
+# manually activate the 'rapids' environment, to cause any filesystem changes made by activation scripts
+RUN . /opt/conda/etc/profile.d/conda.sh; conda activate rapids
+
+# Set more environment variables, to mimic what would happen with 'conda activate rapids'
+#
+# This is a workaround to allow RAPIDS libraries to be accessible to processes that bypass the entrypoint,
+# even though they aren't (and can't be) installed in the 'base' environment.
+# See the discussion in https://github.com/rapidsai/docker/issues/712.
+#
+# This list was generated by building this image without this ENV layer and running:
+#
+#   env > ./old.txt
+#   . /opt/conda/etc/profile.d/conda.sh; conda activate rapids
+#   env > ./new.txt
+#   diff -u ./old.txt ./new.txt
+#
+ENV \
+    CONDA_DEFAULT_ENV=rapids \
+    CONDA_PREFIX=/opt/conda/envs/rapids \
+    CONDA_PREFIX_1=/opt/conda/envs/rapids \
+    CONDA_PREFIX_2=/opt/conda \
+    CONDA_PROMPT_MODIFIER="(rapids)" \
+    CPL_ZIP_ENCODING=UTF-8 \
+    GDAL_DATA=/opt/conda/envs/rapids/share/gdal \
+    GDAL_DRIVER_PATH=/opt/conda/envs/rapids/lib/gdalplugins \
+    GSETTINGS_SCHEMA_DIR=/opt/conda/envs/rapids/share/glib-2.0/schemas \
+    GSETTINGS_SCHEMA_DIR_CONDA_BACKUP= \
+    PATH=/opt/conda/envs/rapids/bin:/opt/conda/condabin:/opt/conda/bin:/usr/local/nvidia/bin:/usr/local/cuda/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
+    PROJ_DATA=/opt/conda/envs/rapids/share/proj \
+    PROJ_NETWORK=ON \
+    XML_CATALOG_FILES="file:///opt/conda/envs/rapids/etc/xml/catalog file:///etc/xml/catalog"
```

Comment:
All of this is to make it *feel like* `conda activate rapids` was run even though it wasn't, in situations where you can't rely on the entrypoint script being run (described by @jacobtomlinson in https://github.com/rapidsai/docker/issues/712#issuecomment-2386370049).

It's a hack and could hopefully be reverted completely in RAPIDS 24.12 through some combination of the following:

* `conda` relaxing its `mamba` constraint and therefore its `fmt` constraint
  - *(e.g. `conda install -y -n base conda 'fmt>=11'` working)*
  - *(see https://github.com/rapidsai/docker/issues/712#issuecomment-2384746639)*
* RAPIDS removing its runtime dependency on `fmt` (https://github.com/rapidsai/build-planning/issues/104)

If reviewers agree with this approach, I'll write up a separate issue to track that work of reverting all of this.

---

NOTE: I'm intentionally not doing this in the `raft-ann-bench` images... those are expected to be used with explicit entrypoints, as far as I can tell, and I've modified all those entrypoints with the appropriate conda activation commands.